kotlin-jvm/src/main/kotlin/playground/NativeDI.kt

```diff
@@ -0,0 +1,225 @@
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+
+
+
+/**
+ * Native Dependency Injection: Pure Kotlin, no framework required Dependency Injection
+ * ideal for smaller-mid level projects which can be migrated easily to Dagger|Hilt like
+ * framewokr
+ *
+ * - [Blog](https://proandroiddev.com/hold-on-before-you-dagger-or-hilt-try-this-simple-di-f674c83ebeec)
+ * - [Github-Sample](https://github.com/ch8n/android-no-3rd-party-mvvm-example)
+ * - [Author](https://chetangupta.net)
+ */
+fun main() {
+
+    val viewModel = Injector.viewModel
+    viewModel.userCallback = { user ->
+        if (user != null) {
+            println(user)
+        }
+    }
+    viewModel.error = { error ->
+        println(error.stackTrace)
+    }
+
+    viewModel.getUser()
+    // for migration guide to Hilt or Dagger, please refer to companion blog
+}
+
+
+/**
+ * Resolver have provider function for construction of dependencies
+ */
+object Resolver {
+
+    fun provideViewModel(sampleRepository: SampleRepository): ViewModel {
+        return ViewModel(sampleRepository)
+    }
+
+    fun provideSampleRepository(localDataSource: DataSourceLocal, remoteDataSource: DataSourceRemote): SampleRepository {
+        return SampleRepositoryImpl(localDataSource, remoteDataSource)
+    }
```

Comment:
It'd be best to put these parameters on separate lines for readability (there's an IDE intention for that if you put the caret at the right place).

Reply:
> For the package, look how it's done for the other samples :)



> For the package, look how it's done for the other samples :)

Thanks, should I do `package playground.native.di` or `package playground.nativedi`